db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,27 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    # only managed hosts should have tokens
+    Token.where(:id => Token.joins("left outer join hosts on hosts.id = tokens.host_id").where("hosts.type != 'Host::Managed'").pluck(:id)).delete_all
```

Comment:
+1 @domcleal, change `pluck(:id)` to `select(:id)`. AFAIK most SQL servers do not allow delete with join syntax (hence subquery is more generic solution).